src/benchq/data_structures/hardware_architecture_models.py

```diff
@@ -24,9 +24,26 @@ class BasicArchitectureModel:
             surface code cycle.
     """
 
+    @property
+    def physical_qubit_error_rate(self) -> float:
+        pass
+
+    @property
+    def surface_code_cycle_time_in_seconds(self) -> float:
+        pass
+
+
+@dataclass(frozen=True)
+class IONTrapModel:
+    physical_qubit_error_rate: float
+    surface_code_cycle_time_in_seconds: float
+
+
+@dataclass(frozen=True)
+class SCModel:
```

Comment:
The this code allows for one to have an SCModel which has variable `physical_qubit_error_rate` and `surface_code_cycle_time_in_seconds`. We should only allow `physical_qubit_error_rate = 1e-4` and `surface_code_cycle_time_in_seconds = 1e-7` or else the widget selectiong gets messed up.

Reply:
Same for the `IONTrapModel`.

Reply:
There are test usages of IONTrapModel and SCModel with different values than 1e-4 and 1e-7. Current approach suggests to use BASIC_* constant dataclasses for this reason.